File: heaps/find_k_closest_elements.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        h = []
        res = []
        for i in arr:
            heappush(h, [abs(i - x), i])
        for _ in range(k):
            res.append(heappop(h)[1])
        return sorted(res)

File: heaps/find_k_closest_elements.rb
require 'algorithms'

def find_closest_elements(arr, k, x)
    res = []
    return res if arr.size < k

    min_heap = Containers::MinHeap.new
    arr.each do |num|
      min_heap.push([((num - x).abs), num])
    end
    
    k.times do
      res << min_heap.pop[1]
    end
    res.sort
end